Extension/src/LanguageServer/extension.ts

```diff
@@ -866,63 +848,82 @@ function onSelectConfiguration(): void {
     }
 }
 
-function onSelectConfigurationProvider(): void {
+async function onSelectConfigurationProvider(): Promise<void> {
     onActivationEvent();
     if (!isFolderOpen()) {
         vscode.window.showInformationMessage(localize("configuration.provider.select.first", 'Open a folder first to select a configuration provider'));
     } else {
-        selectClient().then(client => client.handleConfigurationProviderSelectCommand(), rejected => {});
+        try {
+            const client: Client = await selectClient();
+            client.handleConfigurationProviderSelectCommand();
+        } catch (err) {}
     }
 }
 
-function onEditConfigurationJSON(): void {
+async function onEditConfigurationJSON(): Promise<void> {
     onActivationEvent();
     telemetry.logLanguageServerEvent("SettingsCommand", { "palette": "json" }, undefined);
     if (!isFolderOpen()) {
         vscode.window.showInformationMessage(localize('edit.configurations.open.first', 'Open a folder first to edit configurations'));
     } else {
-        selectClient().then(client => client.handleConfigurationEditJSONCommand(), rejected => {});
+        try {
+            const client: Client = await selectClient();
+            client.handleConfigurationEditJSONCommand();
+        } catch (err) {}
     }
 }
 
-function onEditConfigurationUI(): void {
+async function onEditConfigurationUI(): Promise<void> {
     onActivationEvent();
     telemetry.logLanguageServerEvent("SettingsCommand", { "palette": "ui" }, undefined);
     if (!isFolderOpen()) {
         vscode.window.showInformationMessage(localize('edit.configurations.open.first', 'Open a folder first to edit configurations'));
     } else {
-        selectClient().then(client => client.handleConfigurationEditUICommand(), rejected => {});
+        try {
+            const client: Client = await selectClient();
+            client.handleConfigurationEditUICommand();
+        } catch (err) {}
     }
 }
 
-function onEditConfiguration(): void {
+async function onEditConfiguration(): Promise<void> {
     onActivationEvent();
     if (!isFolderOpen()) {
         vscode.window.showInformationMessage(localize('edit.configurations.open.first', 'Open a folder first to edit configurations'));
     } else {
-        selectClient().then(client => client.handleConfigurationEditCommand(), rejected => {});
+        try {
+            const client: Client = await selectClient();
+            client.handleConfigurationEditCommand();
+        } catch (err) {}
     }
 }
 
-function onGenerateEditorConfig(): void {
+async function onGenerateEditorConfig(): Promise<void> {
     onActivationEvent();
     if (!isFolderOpen()) {
         generateEditorConfig();
     } else {
-        selectClient().then(client => generateEditorConfig(client.RootUri));
+        try {
+            const client: Client = await selectClient();
+            generateEditorConfig(client.RootUri);
```

Comment:
This call to selectClient() did not have a rejected => {}, so it would have thrown an exception.  Also, any exception from generateEditorConfig() should be allowed to propagate.